app/server.jsx

```diff
@@ -91,19 +92,18 @@ export default function render(req, res) {
     } else if (props) {
       // This method waits for all render component
       // promises to resolve before returning to browser
-      preRenderMiddleware(
-        store.dispatch,
-        props.components,
-        props.params
-      )
-      .then(() => {
-        const initialState = store.getState();
+      store.dispatch({ type: types.CREATE_REQUEST });
+      preRenderMiddleware(props)
```

Comment:
This bit is duplicated in both client and server, but I think it's better than making `preRenderMiddleware` do everything.